### ISSUE TITLE
knife vault refresh broken - chefdk0.7.0/chef11.1.1

### ISSUE THREAD

**Issue by karcaw** [chef/chef-vault#182]
knife vault refresh stopped working at some point.  It runs, but puts a client key with a blank key in the databag.

knife exec -E "require 'chef-vault';client = ChefVault::ChefPatch::ApiClient.load('<client>');puts client"
returns:
client[]

knife exec -E "require 'chef-vault';client = ChefVault::ChefPatch::ApiClient.load('<client>');puts client.name"
returns a blank line,

knife exec -E "require 'chef-vault';client = ChefVault::ChefPatch::ApiClient.load('<client>');puts client.public_key"
shows the key.

if i change things in the api_client.rb patch so that it picks up 'name' instead of 'clientname' things seem to work better.

**Comment by ambientsound** [comment]
I'm having this issue with ChefDK 0.10.0, and can confirm that @karcaw's solution works:

```
$ chef -v
Chef Development Kit Version: 0.10.0
chef-client version: 12.5.1
berks version: 4.0.1
kitchen version: 1.4.2
```

Running this command fixes the problem:

```
sudo sed -i -e 's/clientname/name/' /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/chef_patch/api_client.rb
```

**Comment by ambientsound** [comment]
Also, it is not only a problem with refresh, but also with update, create, and all other commands that sets the search string.

**Comment by schredder** [comment]
I can also confirm I had this same issue, and the solution worked for me too.
I have chef 12.5.1 installed in ruby 2.1.6 with rbenv.

**Comment by thommay** [comment]
Should be fixed by #184
